seed_data/lib.py

```diff
@@ -250,25 +253,36 @@ def set_data(self, obj, course_run, grade=DEFAULT_GRADE, **kwargs):
 }
 
 
-def clear_edx_data_for_course_run(user, course_run, models=None):
-    """Clears all of a User's cached edX data for a CourseRun, or just the data for specific cached models"""
-    if models:
-        handlers = [CACHED_HANDLERS[model] for model in models]
-    else:
-        handlers = CACHED_HANDLERS.values()
-    for cached_model_handler in handlers:
-        cached_model_handler(user).clear_all(course_run=course_run)
-
-
-def clear_edx_data_for_course(user, course, models=None):
-    """Clears all of a User's cached edX data for a Course, or just the data for specific cached models"""
+def ensure_cached_edx_data_for_user(user):
+    """
+    Ensures that all possible cached edX records have been initialized for a User
+    """
+    all_course_runs = CourseRun.objects.filter(course__program__live=True).exclude(
+        Q(edx_course_key__isnull=True) | Q(edx_course_key__exact='')
+    ).all()
+    all_course_run_ids = set([course_run.id for course_run in all_course_runs])
```

Comment:
Nitpick: you don't need those braces at all. You can use a generator comprehension, like this:
```python
set(course_run.id for course_run in all_course_runs)
```
It's faster too, because it will turn the query results directly into a set, instead of turning them into a list first.